tests/cmdline/commands/test_group.py

```diff
@@ -134,24 +134,146 @@ def test_delete(self, run_cli_command):
+            raises=True,
+            use_subprocess=True,
+        )
+        result = run_cli_command(
+            cmd_group.group_delete,
+            ['--force', 'group_test_delete_01', '--past-days', '1', 'label==group_test_delete_07'],
+            raises=True,
+            use_subprocess=True,
+        )
+        result = run_cli_command(
+            cmd_group.group_delete,
+            ['--force', 'group_test_delete_01', '--startswith', 'non_existent', 'label==group_test_delete_07'],
+            raises=True,
+            use_subprocess=True,
+        )
+        result = run_cli_command(
+            cmd_group.group_delete,
+            ['--force', 'group_test_delete_01', '--endswith', 'non_existent', 'label==group_test_delete_07'],
+            raises=True,
+            use_subprocess=True,
+        )
+        result = run_cli_command(
+            cmd_group.group_delete,
+            ['--force', 'group_test_delete_01', '--contains', 'non_existent', 'label==group_test_delete_07'],
+            raises=True,
+            use_subprocess=True,
+        )
+        result = run_cli_command(
+            cmd_group.group_delete,
+            ['--force', 'group_test_delete_01', '--node', 'non_existent', 'label==group_test_delete_07'],
+            raises=True,
+            use_subprocess=True,
+        )
+
+        # 9) --user should delete groups for a specific user
+        #   --all-users should delete groups for all users
+        user1 = orm.User(email='user1@example.com')
+        user2 = orm.User(email='user2@example.com')
+        user3 = orm.User(email='user3@example.com')
+        user1.store()
+        user2.store()
+        user3.store()
+
+        orm.Group(label='group_test_delete_08', user=user1).store()
+        orm.Group(label='group_test_delete_09', user=user2).store()
+        orm.Group(label='group_test_delete_10', user=user3).store()
+
+        result = run_cli_command(cmd_group.group_delete, ['--force', '--user', user1.email], use_subprocess=True)
+        with pytest.raises(exceptions.NotExistent):
+            orm.load_group(label='group_test_delete_08')
+
+        result = run_cli_command(cmd_group.group_delete, ['--force', '--all-users'], use_subprocess=True)
+        with pytest.raises(exceptions.NotExistent):
+            orm.load_group(label='group_test_delete_09')
+        with pytest.raises(exceptions.NotExistent):
+            orm.load_group(label='group_test_delete_10')
+
+        # 10) --startswith, --endswith, --contains should delete groups with labels that match the filter
+        orm.Group(label='START_13').store()
+        orm.Group(label='14_END').store()
+        orm.Group(label='contains_SOMETHING_').store()
+
+        result = run_cli_command(cmd_group.group_delete, ['--force', '--startswith', 'START'], use_subprocess=True)
+        with pytest.raises(exceptions.NotExistent):
+            orm.load_group(label='START_13')
```

Comment:
Either check the output to make sure that _only_ this group is deleted or verify that the other groups haven't been deleted explicitly.

Reply:
Ok thanks, 
Now I check the output, I think it's faster than loading other groups.